pkg/controller/config/config_controller_test.go

```diff
@@ -0,0 +1,206 @@
+	securityv1 "github.com/openshift/api/security/v1"
+	fakesecurityclient "github.com/openshift/client-go/security/clientset/versioned/fake"
+	op "github.com/tektoncd/operator/pkg/apis/operator/v1alpha1"
+	"github.com/tektoncd/operator/pkg/flag"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func TestConfigControllerReplaceImages(t *testing.T) {
+	t.Run("for_pipelines", func(t *testing.T) {
+		var (
+			configName = "cluster"
+			namespace  = "openshift-pipelines"
+			deployment = "tekton-pipelines-controller"
+			container  = "tekton-triggers-controller"
+			image      = "registry.redhat.io/osbs/controller:latest"
+			arg        = "-shell-image"
+			argImage   = "registry.redhat.io/osbs/ubi8:latest"
+		)
+
+		// GIVEN
+		os.Setenv("IMAGE_PIPELINE_"+strings.ToUpper(container), image)
+		os.Setenv("IMAGE_PIPELINE_"+strings.ToUpper(arg), argImage)
+		config := configFor(configName, namespace)
+		cl := feedConfigMock(config)
+		secCl := feedSCCMock("privileged")
+		pipelines, err := mfFor("pipelines", cl)
+		assetNoEror(err, "failed to create manifestival for pipelines;", t)
+		req := requestFor(configName, namespace)
+		r := ReconcileConfig{scheme: scheme.Scheme, client: cl, pipeline: pipelines, secClient: secCl}
+
+		// WHEN
+		_, err = r.applyPipeline(req, config)
+
+		// THEN
+		assetNoEror(err, "failed to reconcile for applyPipeline;", t)
+		assertContainerHasImage(deployment, container, image, r.client, t)
+		assertContainerArgHasImage(deployment, arg, argImage, r.client, t)
+	})
+
+	t.Run("for_triggers_addon", func(t *testing.T) {
+		var (
+			configName = "cluster"
+			namespace  = "openshift-pipelines"
+			deployment = "tekton-triggers-controller"
+			container  = "tekton-triggers-controller"
+			image      = "registry.redhat.io/osbs/controller:latest"
+			arg        = "-el-image"
+			argImage   = "registry.redhat.io/osbs/eventlistenersink:latest"
+		)
+
+		// GIVEN
+		os.Setenv("IMAGE_TRIGGERS_"+strings.ToUpper(container), image)
+		os.Setenv("IMAGE_TRIGGERS_"+strings.ToUpper(arg), argImage)
+		config := configFor(configName, namespace)
+		cl := feedConfigMock(config)
+		addons, err := mfFor("addons", cl)
+		assetNoEror(err, "failed to create manifestival for triggers addons;", t)
+		req := requestFor(configName, namespace)
+		r := ReconcileConfig{scheme: scheme.Scheme, client: cl, addons: addons}
+
+		// WHEN
+		_, err = r.applyAddons(req, config)
+
+		// THEN
+		assetNoEror(err, "failed to reconcile for applyPipeline;", t)
+		assertContainerHasImage(deployment, container, image, r.client, t)
+		assertContainerArgHasImage(deployment, arg, argImage, r.client, t)
+	})
+}
+
+func configFor(name string, namespace string) *op.Config {
```

Comment:
how about newConfig?

Reply:
as a rule of thump all function that uses `For` should be of the form `xForY` e.g. `priceForApple`